fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -357,8 +359,17 @@ public void execute(TUniqueId queryId) throws Exception {
                         context.getState().setError(e.getMysqlErrorCode(), e.getMessage());
                         return;
                     }
-                    // limitations: partitionNum, tabletNum, cardinality
+                    // limitations: partition_num, tablet_num, cardinality
+                    List<ScanNode> scanNodeList = planner.getScanNodes();
+                    for (ScanNode scanNode : scanNodeList) {
+                        if (scanNode instanceof OlapScanNode) {
+                            OlapScanNode olapScanNode = (OlapScanNode) scanNode;
+                            Catalog.getCurrentCatalog().getSqlBlockRuleMgr().checkLimitaions(olapScanNode.getSelectedPartitionNum().longValue(),
+                                        olapScanNode.getSelectedTabletsNum(), olapScanNode.getCardinality(), analyzer.getQualifiedUser());
```

Comment:
I think analyzer.getQualifiedUser() can be replaced by context.getQualifiedUser()

Reply:
Ok, I will fix it in my PR for a Bugfix.